src/Agent.Listener/Diagnostics/MtuInfo.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Net.NetworkInformation;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Diagnostics
+{
+    class MtuInfo : IDiagnosticInfo
+    {
+        public void Execute(ITerminal terminal)
+        {
+            NetworkInterface[] nics = NetworkInterface.GetAllNetworkInterfaces();
+            IPGlobalProperties properties = IPGlobalProperties.GetIPGlobalProperties();
+            terminal.WriteLine(string.Format("IP interface information for {0}.{1}", properties.HostName, properties.DomainName));
+            terminal.WriteLine();
+
+            foreach (NetworkInterface adapter in nics)
+            {
+                terminal.WriteLine(adapter.Description);
+
+                if (adapter.Supports(NetworkInterfaceComponent.IPv4))
+                {
+                    IPInterfaceProperties adapterProperties = adapter.GetIPProperties();
```

Comment:
can adapterProperties be null?

Reply:
Not according to docs.  There is an exception catch around this test if the contract ever did change.